### PR TITLE
Refactor scheduler server

### LINKED ISSUE #1908: [Ballista] Refactor ballista scheduler server
**Is your feature request related to a problem or challenge? Please describe what you are trying to do.**

Currently the ballista scheduler mod messes everything up in the lib file, which is not good for code reviewing.

**Describe the solution you'd like**

It's better to reorganize the scheduler contents as following:
- external_scaler, for exposing scheduler internal metrics
- grpc, for grpc service 
- event_loop, for dealing with internal event-based state changes (TODO) #1909 

**Additional context**
#1704

### PR DESCRIPTION
**Pull request by yahoNanJing**
# Which issue does this PR close?

Closes #1908.

 # Rationale for this change

Reorganize the scheduler contents for better code review and easy feature additions.

# What changes are included in this PR?

The following separate mod are extracted from the scheduler contents:
- external_scaler, for exposing scheduler internal metrics
- grpc, for grpc service

# Are there any user-facing changes?

### CODE REVIEW

File: ballista/rust/scheduler/src/scheduler_server/mod.rs [hunk 1]
```diff
@@ -0,0 +1,294 @@
...
+        job_id: &str,
+    ) -> Result<(Vec<Vec<TaskDefinition>>, usize), BallistaError> {
+        let mut ret: Vec<Vec<TaskDefinition>> =
+            Vec::with_capacity(available_executors.len());
+        for _idx in 0..available_executors.len() {
+            ret.push(Vec::new());
+        }
+        let mut num_tasks = 0;
+        loop {
+            info!("Go inside fetching task loop");
```
Comment (thinkharderdev) [inline ballista/rust/scheduler/src/scheduler_server/mod.rs:176]: realize you didn't add this here, but can we change this to debug level? could add a lot of noise to the logs

**Review by thinkharderdev — APPROVED** [review]
Looks great!

**Comment by alamb** [comment]
Rather than allow PRs to accumulate, I'll merge them in as they seem uncontroversial and have been reviewed. We can always continue reorganizing as we move forward. 

Thanks @yahoNanJing  for the code and @thinkharderdev  for the review

**Comment by alamb** [comment]
cc @realno  @liukun4515

**Comment by liukun4515** [comment]
> cc @realno @liukun4515

I go through this pull request, and it's just split the code into some separated files.
looks good to me.

**Pull request merged**